bazel/foreign_cc/BUILD

```diff
@@ -0,0 +1,82 @@
+licenses(["notice"])  # Apache 2
+
+load("//bazel:envoy_build_system.bzl", "envoy_package")
+load("@rules_foreign_cc//tools/build_defs:cmake.bzl", "cmake_external")
+load("@foreign_cc_platform_utils//:tools.bzl", "NINJA_COMMAND", "NINJA_DEP")
+
+envoy_package()
+
+cmake_external(
+    name = "ares",
+    cache_entries = {
```

Comment:
Add `"CMAKE_BUILD_TYPE":  "RelWithDebInfo"` (possibly `"Debug"` for Windows builds?)

Reply:
Done, but I'm going to skip the Windows parts and let @sesmith177 take this.